cylc/flow/scripts/workflow_state.py

```diff
@@ -18,252 +18,409 @@
+        if not self._db_checker:
             try:
-                self.checker = CylcWorkflowDBChecker(
-                    self.args['run_dir'], self.args['workflow_id'])
-                connected = True
-                # ... but ensure at least one poll after connection:
-                self.n_polls -= 1
+                self._db_checker = CylcWorkflowDBChecker(
+                    get_cylc_run_dir(self.alt_cylc_run_dir),
+                    self.workflow_id
+                )
             except (OSError, sqlite3.Error):
-                if self.n_polls >= max_polls:
-                    raise
-                if cylc.flow.flags.verbosity > 0:
-                    sys.stderr.write('.')
-                sleep(self.interval)
-        if cylc.flow.flags.verbosity > 0:
-            sys.stderr.write('\n')
-
-        if connected and self.args['cycle']:
-            try:
-                fmt = self.checker.get_remote_point_format()
-            except sqlite3.OperationalError as exc:
-                try:
-                    fmt = self.checker.get_remote_point_format_compat()
-                except sqlite3.OperationalError:
-                    raise exc  # original error
-            if fmt:
-                my_parser = TimePointParser()
-                my_point = my_parser.parse(self.args['cycle'], dump_format=fmt)
-                self.args['cycle'] = str(my_point)
-        return connected, self.args['cycle']
-
-    async def check(self):
-        """Return True if desired workflow state achieved, else False"""
-        return self.checker.task_state_met(
-            self.args['task'], self.args['cycle'],
-            self.args['status'], self.args['message'])
+                LOG.debug("DB not connected")
+                return None
+
+        return self._db_checker
+
+    async def check(self) -> bool:
+        """Return True if requested state achieved, else False.
+
+        Called once per poll by super() so only find and connect once.
+
+        Store self.result for external access.
+
+        """
+        if self.workflow_id is None and not self._find_workflow():
+            return False
+
+        if self.db_checker is None:
+            return False
+
+        if self.cycle is None:
+            # Adjust target cycle point to the DB format.
+            self.cycle = self.db_checker.adjust_point_to_db(
+                self.cycle_raw, self.offset)
+
+        self.result = self.db_checker.workflow_state_query(
+            self.task, self.cycle, self.selector, self.is_trigger,
+            self.is_message, self.flow_num
+        )
+        if self.result:
+            # End the polling dot stream and print inferred runN workflow ID.
+            self.db_checker.display_maps(
+                self.result, self.old_format, self.pretty_print)
+
+        return bool(self.result)
 
 
 def get_option_parser() -> COP:
     parser = COP(
         __doc__,
-        argdoc=[WORKFLOW_ID_ARG_DOC]
+        argdoc=[ID_MULTI_ARG_DOC]
```

Comment:
Why multi, when it's only supposed to take 1 ID?

Reply:
Because WORKFLOW_ID_ARG_DOC is for workflow, not workflow/task.

I've now created a new doc string, for single ID.